### ISSUE TITLE
compilation error in Windows

### ISSUE THREAD

**Issue by AlejandroCornejo** [KratosMultiphysics/Kratos#2570]
I've merged my branch with the master one and I got this compilation error (I suppose only in windows because @loumalouomega has compiled in Linux without any problem).

The error message is the following one:
![imagen](https://user-images.githubusercontent.com/26402605/43203911-c52e1972-901f-11e8-969b-cfcff05c60a2.png)
 
in text format:
'amgcl::make_solver<amgcl::preconditioner::schur_pressure_correction<USolver,PSolver>,amgcl::runtime::solver::wrapper<Backend,amgcl::solver::detail::default_inner_product>>::make_solver(amgcl::make_solver<amgcl::preconditioner::schur_pressure_correction<USolver,PSolver>,amgcl::runtime::solver::wrapper<Backend,amgcl::solver::detail::default_inner_product>> &&)': el argumento 2 no puede convertirse de 'boost::property_tree::ptree' a 'const amgcl::make_solver<amgcl::preconditioner::schur_pressure_correction<USolver,PSolver>,amgcl::runtime::solver::wrapper<Backend,amgcl::solver::detail::default_inner_product>>::params &'	Kratos	d:\masterkratos\kratos\linear_solvers\amgcl_ns_solver.h	251	

@roigcarlo

**Comment by roigcarlo** [comment]
Hi @AlejandroCornejo ,

I've just updated but can't reproduce the problem (also it dosn't seem to fail in appveyor), can you try to make a clean compilation?

**Comment by AlejandroCornejo** [comment]
I've Rebuild everything and obtain the same... Do i should download the code from 0?

**Comment by roigcarlo** [comment]
Honestly Idk. It also could be caused by the version of the boost libs. Which one are you using?

**Comment by AlejandroCornejo** [comment]
I use the 1.59 version but up to now it was working properly...

**Comment by loumalouomega** [comment]
@ddemidov uses many new features from the boost libraries, maybe he can confirm the minimal version required, ...and we can update the Wiki
In any case update boost now is easy, after removing boost::python it is not necessary to compile again

**Comment by roigcarlo** [comment]
I am using 1.67 myself. If the problem is the library I can digg out which is the new minimal.

**Comment by ddemidov** [comment]
Is it possible to get the full compile log? The errors posted do not mean anything I think.

Re boost version: part of the upgrade was to get rid of as much boost as possible (replacing it with C++11 features). I think at this point the only boost library amgcl depends on is header-only Boost.property_tree, and I don't think it matters which boost version is used as long as it's not too ancient.

**Comment by ddemidov** [comment]
It looks like this version of `params()` constructor is not active (may be not in this exact file, but all `params()` in amgcl are structured in a similar way):

https://github.com/KratosMultiphysics/Kratos/blob/a618f8dfc754114d86cd69eb5168fe0adb358939/external_libraries/amgcl/solver/bicgstab.hpp#L88-L104

which is strange, since `<boost/property_tree/ptree.hpp>` is included before amgcl [here](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/linear_solvers/amgcl_solver.h#L40-L53) and it should bring BOOST_VERSION along.

EDIT: could this be a result of merge gone wrong?

**Comment by ddiezrod** [comment]
Just to mention that I had a similar error in gcc using boost 1.59. I could compile the master branch only after upgrading boost to 1.67.

**Comment by loumalouomega** [comment]
Nice, now we should detect which version is the minimum in order to update the wiki

**Comment by roigcarlo** [comment]
I've already updated the wiki with the 1.67. If we detect that is possible to compiler with a lower version I'll change it again.

**Comment by roigcarlo** [comment]
@ddemidov I am investigating what could have happen with the BOOST_VERSION. It seems that it's also giving problems in the nightly but not in travis for example.

Maybe something is being redefined at some point

**Comment by ddemidov** [comment]
Can you test if adding `#include <boost/version.hpp>` [here](https://github.com/KratosMultiphysics/Kratos/blob/2841760d8272df9ed0e7402f9c923f1ca926d67e/kratos/linear_solvers/amgcl_solver.h#L40) and [here](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/linear_solvers/amgcl_ns_solver.h#L38) helps with boost 1.59? I think it is possible `<boost/property_tree/ptree.hpp>` does not do that in 1.59.

**Comment by roigcarlo** [comment]
Ok, let me try.

For the record, On travis CI we are using boost 1.54. But we have unitary builds enabled there, so probably its ignoring missing includes since it includes it all together

**Comment by ddemidov** [comment]
If that helps, I think I'll enable boost.property_tree constructors by default and let users disable it on demand (to stay completely boost-free), and come up with a new PR.

**Comment by AlejandroCornejo** [comment]
I've tried to include those files @ddemidov  and now it compiles! 

Thank you,

**Comment by roigcarlo** [comment]
Works for me too :+1:

**Comment by ddemidov** [comment]
Ok, a PR is incoming.
